lib/Dialect/TritonGPU/Transforms/RemoveLayoutConversions.cpp

```diff
@@ -137,11 +137,12 @@ class SimplifyReduceCvt : public mlir::RewritePattern {
           op->getLoc(), newTy, newOperands[i]);
     }
 
+    rewriter.setInsertionPoint(reduce);
```

Comment:
Because we're matching on the `convert_layout` op, the rewriter inserts directly after that op and potentially before the rest of the arguments are ready.